core/service/manager.go

```diff
@@ -115,30 +106,31 @@ func (manager *Manager) Start(options Options) (err error) {
 
 	manager.discovery.Start(providerID, proposal)
 
-	err = manager.service.Serve(providerID)
+	err = service.Serve(providerID)
+	if err != nil {
+		return err
+	}
+	manager.servicePool.Add(service)
+
 	manager.discovery.Wait()
 	return err
 }
 
 // Kill stops service
 func (manager *Manager) Kill() error {
-	var errDialogWaiter, errService error
+	errStop := utils.ErrorCollection{}
 
 	if manager.discovery != nil {
 		manager.discovery.Stop()
 	}
 	if manager.dialogWaiter != nil {
-		errDialogWaiter = manager.dialogWaiter.Stop()
+		if err := manager.dialogWaiter.Stop(); err != nil {
```

Comment:
I'd move the `if err != nil` condition to `utils.ErrorCollection.Add()`  to make the code simple:

```golang
errStop.Add(manager.dialogWaiter.Stop())
```

Reply:
Done